src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,80 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.exceptions.CommandParamException;
+
+/**
+ * This is a Deadline class that extends Task.
+ */
+public class Deadline extends Task {
+
+    /**
+     * These are class field of Deadline.
+     */
+    protected LocalDate date;
+    protected LocalTime time;
+
+    /**
+     * This is a Deadline Constructor.
+     *
+     * @param description A String representing the description of the deadline.
+     * @param by A String representing the date and time of deadline.
+     * @throws CommandParamException An exception thrown when the date and time is not formatted properly.
+     */
+    public Deadline(String description, String by) throws CommandParamException {
+        super(description);
+        String[] dateTime = by.trim().split(" ");
+        //index 0 is date, 1 is time
```

Comment:
you could leave a space between `//` and the comment, but otherwise a good use of comments 👍